lua/nvim-tree.lua

```diff
@@ -389,8 +389,7 @@ local function setup_autocommands(opts)
       pattern = "NvimTree_*",
       callback = function()
         vim.schedule(function()
```

Comment:
but how would that be better than `norm! zz`? isn't the norm solution strictly simpler?

Reply:
@emmanueltouzery, what was missing from my implementation is ensuring that `feedkeys()` only fires in the specified buffer, so `nvim_buf_call()` should be called _regardless_.

I wasn't able to reproduce the issue, but using the `feedkeys()` vs `vim.cmd` could be debatable.

Reply:
ok, i'll try your solution asap, but most likely i won't manage today :+1:

Reply:
Thanks @kylo252

Reply:
so I tried @kylo252 's solution and.. it didn't work for me.

this is the test startup file for @kylo252 's solution:
```lua
vim.cmd [[set runtimepath=$VIMRUNTIME]]
vim.cmd [[set packpath=/tmp/nvt-min/site]]
local package_root = "/tmp/nvt-min/site/pack"
local install_path = package_root .. "/packer/start/packer.nvim"
local function load_plugins()
  require("packer").startup {
    {
      "wbthomason/packer.nvim",
      {"emmanueltouzery/nvim-tree.lua", branch = 'zz_kylo'},
      "nvim-tree/nvim-web-devicons",
      -- ADD PLUGINS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
    },
    config = {
      package_root = package_root,
      compile_path = install_path .. "/plugin/packer_compiled.lua",
      display = { non_interactive = true },
    },
  }
end
if vim.fn.isdirectory(install_path) == 0 then
  print "Installing nvim-tree and dependencies."
  vim.fn.system { "git", "clone", "--depth=1", "https://github.com/wbthomason/packer.nvim", install_path }
end
load_plugins()
require("packer").sync()
vim.cmd [[autocmd User PackerComplete ++once echo "Ready!" | lua setup()]]
vim.opt.termguicolors = true
vim.opt.cursorline = true

-- MODIFY NVIM-TREE SETTINGS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
_G.setup = function()
  require("nvim-tree").setup {
    view = {
      centralize_selection = true,
    }
  }
  vim.cmd("NvimTreeToggle")
end

function _G.repro()
  vim.cmd("wincmd l")
  vim.cmd("NvimTreeFocus")
  vim.fn.input("what say you?")
end
```
and this is for my solution:
```lua
vim.cmd [[set runtimepath=$VIMRUNTIME]]
vim.cmd [[set packpath=/tmp/nvt-min/site]]
local package_root = "/tmp/nvt-min/site/pack"
local install_path = package_root .. "/packer/start/packer.nvim"
local function load_plugins()
  require("packer").startup {
    {
      "wbthomason/packer.nvim",
      {"emmanueltouzery/nvim-tree.lua", branch = 'no_feed_keys'},
      "nvim-tree/nvim-web-devicons",
      -- ADD PLUGINS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
    },
    config = {
      package_root = package_root,
      compile_path = install_path .. "/plugin/packer_compiled.lua",
      display = { non_interactive = true },
    },
  }
end
if vim.fn.isdirectory(install_path) == 0 then
  print "Installing nvim-tree and dependencies."
  vim.fn.system { "git", "clone", "--depth=1", "https://github.com/wbthomason/packer.nvim", install_path }
end
load_plugins()
require("packer").sync()
vim.cmd [[autocmd User PackerComplete ++once echo "Ready!" | lua setup()]]
vim.opt.termguicolors = true
vim.opt.cursorline = true

-- MODIFY NVIM-TREE SETTINGS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
_G.setup = function()
  require("nvim-tree").setup {
    view = {
      centralize_selection = true,
    }
  }
  vim.cmd("NvimTreeToggle")
end

function _G.repro()
  vim.cmd("wincmd l")
  vim.cmd("NvimTreeFocus")
  vim.fn.input("what say you?")
end
```

I'm not sure how to process, @kylo252, maybe you can test this and prepare a patch that fixes that issue? Note that this also fails in my real-world scenario, not only in this "artificial" reproduction.

Reply:
again. run the test files with `nvim -nu filename.lua`.

Reply:
And again, the instructions to reproduce are:

run it with nvim -nu repro.lua
let it launch, set up, then once nvim-tree is opened, run: :lua repro().

Without the fix, at the "What say you?" prompt, zz will be inserted.
With the fix, there'll be no zz.

Reply:
I guess I didn't account for the use of `input()` 😅 

we can go with this version then

```lua
    callback = function()
      local bufnr = api.nvim_get_current_buf()
      vim.schedule(function()
        api.nvim_buf_call(bufnr, function()
          vim.cmd [[norm! zz]]
        end)
      end)
    end,
```

Reply:
yes, that works! force-pushed so that's now what the MR does.

btw it wasn't only a problem with input, in fact my original issue was with the "lightspeed" plugin, which waits for user input, and was "fed" that zz, therefore leading to unpredictable behavior.